ui/src/components/core/icon/icon.styles.ts

```diff
@@ -0,0 +1,43 @@
+import { dripStitches } from '../../../theme/stitches'; //TODO replace for absolute path
+
+const { styled } = dripStitches;
+
+export abstract class IconStyles {
+  static readonly Container = styled('span', {
+    transition: 'transform $default',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+
+    variants: {
+      size: {
+        sm: {
+          fontSize: '$md',
+        },
+        md: {
+          fontSize: '$2xl',
+        },
+        lg: {
+          fontSize: '$3xl',
+        },
+      },
+
+      rotate: {
+        true: {
+          transform: 'rotate(-180deg)',
+        },
+      },
+    },
+
+    defaultVariants: {
+      size: 'md',

```

Comment:
I would leave the font size to `font-size: 1em` instead of defining a default size for it, so it would get the size of the parent so it fits better were it is being display instead you set a size.